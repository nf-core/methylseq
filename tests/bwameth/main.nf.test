import com.askimed.nf.test.util.FileUtil;

nextflow_pipeline {

    name "bwameth"
    script "main.nf"

    test("Single End") {
        when {
            params {
                aligner = "bwameth"
                save_reference = true
                outdir = "$outputDir"
            }
        }

        then {
            // Clean up anything from previous runs
            FileUtil.deleteDirectory(new File(".nf-test/indexes/"));
            new File(".nf-test/indexes/").mkdirs()
            // Copy the ref genome to a stable location for next tests
            FileUtil.copyDirectory(
                new File("$outputDir/bwameth/reference_genome/bwameth/").getAbsolutePath(),
                new File(".nf-test/indexes/bwameth/").getAbsolutePath()
            )

            assert workflow.success
            assert snapshot(workflow.trace.tasks().size()).match()
            assert new File("$outputDir/multiqc/bwameth/multiqc_report.html").exists()
            assert new File("$outputDir/bwameth/deduplicated/SRR389222_sub1.markdup.sorted.bam").exists()
            assert new File("$outputDir/bwameth/reference_genome/bwameth/genome.fa.bwameth.c2t").exists()
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End with index") {
        when {
            params {
                aligner = "bwameth"
                // Generated by previous test
                bwa_meth_index = ".nf-test/indexes/bwameth/"
                methyl_kit = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.trace.tasks().size()).match()
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End RRBS") {
        when {
            params {
                aligner = "bwameth"
                save_reference = true
                skip_trimming = true
                rrbs = true
                save_reference = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.trace.tasks().size()).match()
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End RRBS with index") {
        when {
            params {
                aligner = "bwameth"
                skip_trimming = true
                rrbs = true
                // Generated by previous test
                bwa_meth_index = ".nf-test/indexes/bwameth/"
                methyl_kit = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.trace.tasks().size()).match()
            // FIXME assert snapshot(workflow).match()
        }
    }

}
