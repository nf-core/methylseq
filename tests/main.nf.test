import com.askimed.nf.test.util.FileUtil;

nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "methylseq"

    test("METHYLSEQ Workflow main.nf") {

        when {
            params {
                save_reference = true
                outdir = "$outputDir"
            }
        }

        then {
            // Assert Workflow Success
            assert workflow.success

            // Bismark Methylation Calls Splitting Report
            assert snapshot(path("$outputDir/bismark/methylation_calls/splitting_report/Ecoli_10K_methylated_1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt"),
                            path("$outputDir/bismark/methylation_calls/splitting_report/SRR389222_sub1_trimmed_bismark_bt2.deduplicated_splitting_report.txt"),
                            path("$outputDir/bismark/methylation_calls/splitting_report/SRR389222_sub2_trimmed_bismark_bt2.deduplicated_splitting_report.txt")).match("splitting_report")

            // Bismark Summary Report
            assert snapshot(path("$outputDir/bismark/summary/bismark_summary_report.txt")).match("summary")

            // Reference Genome
            assert snapshot(path("$outputDir/bismark/reference_genome/BismarkIndex/genome.fa")).match("reference_genome")

            // Qualimap Report
            assert snapshot(path("$outputDir/qualimap/Ecoli_10K_methylated/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub1/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub2/genome_results.txt")).match("qualimap")

            // Software Versions
            assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions")
        }
    }

}

