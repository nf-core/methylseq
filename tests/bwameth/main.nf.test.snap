{
    "single_end": {
        "content": [
            "Ecoli_10K_methylated.flagstat:md5,f520e87cc4050799a0dbed72909572ec",
            "SRR389222_sub1.flagstat:md5,ad8159a7683b9c3057bf54ccb77a2ddc",
            "SRR389222_sub2.flagstat:md5,2166a52f81f9b3881e8ac060876aa2d9",
            "Ecoli_10K_methylated.markdup.sorted_CpG.bedGraph:md5,d3bce01078b6e7749f3906bf9faad5b5",
            "SRR389222_sub1.markdup.sorted_CpG.bedGraph:md5,0944ad0e3b05791bf6dfd06be7ca4af1",
            "SRR389222_sub2.markdup.sorted_CpG.bedGraph:md5,0a3c19af55d1625e91903ae25d1d2098",
            "genome_results.txt:md5,1ee2e29ac41c2cd7e6a0f00741e8d16e",
            "genome_results.txt:md5,2c755b15ae5ad4a1d29395320a6264ca",
            "genome_results.txt:md5,3c7d14ea070df510941c55f72706a7a0",
            "software_versions.yml:md5,e08c216cc6de759f0a6dc82d96a9a014"
        ],
        "timestamp": "2023-04-25T16:44:12+0000"
    },
    "single_end_rrbs": {
        "content": [
            "Ecoli_10K_methylated.flagstat:md5,7c3bae0f9c700dde88785a7e94c6ae2c",
            "SRR389222_sub1.flagstat:md5,897d500a710a56a7098172167fa71108",
            "SRR389222_sub2.flagstat:md5,2d680c5657fc029b703b217be4af68e2",
            "Ecoli_10K_methylated.sorted_CpG.bedGraph:md5,3656c54896013b76de978ab95c3f5975",
            "SRR389222_sub1.sorted_CpG.bedGraph:md5,4475a9462ce86a019015b69b82a9d5a2",
            "SRR389222_sub2.sorted_CpG.bedGraph:md5,18b8845d9d985d3c203ca6c9c40aab1b",
            "genome_results.txt:md5,ca30867e16c5ca25c850c81d88563d41",
            "genome_results.txt:md5,f01c68dd46ca9ec0af3d6ec926763b75",
            "genome_results.txt:md5,f4513f5aa1622937ccdcfe92a6f877c0",
            "software_versions.yml:md5,206a9e52256fca71c3ca921ac8ef78f7"
        ],
        "timestamp": "2023-04-25T16:44:12+0000"
    }
}